src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -614,6 +615,10 @@ private User impersonate(final SecurityRequest request, final User originalUser)
             // loop over all http/rest auth domains
             for (final AuthDomain authDomain : restAuthDomains) {
                 final AuthenticationBackend authenticationBackend = authDomain.getBackend();
+                // Skip over the OnBehalfOfAuthenticator since it is not compatible for user impersonation
+                if (authDomain.getHttpAuthenticator() instanceof OnBehalfOfAuthenticator) {
```

Comment:
I know this is the case on main too, but this should probably be formalized as a property of the auth domain? As a default implementation the authDomain could have:

```
default boolean supportsImpersonation() { return true; }
```

and any authDomain that does not support it can override:

```
@Override
public boolean supportsImpersonation() { return false; }
```